### PR TITLE
feat: Add optional configuration to disable Lineage list view links

### PR DESCRIPTION
**Pull request by ozandogrultan**
Signed-off-by: Ozan Dogrultan <ozan.dogrultan@deliveryhero.com>

### Summary of Changes

This PR adds a new optional configuration to existing `TableLineageConfig`, to set up field based rules, which are used to disable "lineage list view" links based on regular expression matching. This config is particularly useful when some of the table resources should appear in the Lineage graph, however are not ingested into catalog or should not be accessible by users.

### Tests

No test changes are required.

### Documentation

A comment added to the new config explaining the purpose.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/amundsen_application/static/js/components/ResourceListItem/TableListItem/index.tsx [hunk 1]
```diff
@@ -34,10 +35,12 @@ const TableListItem: React.FC<TableListItemProps> = ({
   table,
   logging,
   tableHighlights,
+  disabled,
 }) => (
   <li className="list-group-item clickable">
     <Link
       className="resource-list-item table-list-item"
+      style={{ pointerEvents: disabled ? 'none' : '' }}
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/ResourceListItem/TableListItem/index.tsx:43]: We prefer not to use the style prop.
I suggest to conditionally add a class, something like 'is-disabled', and add this styles into the corresponding stylesheet.

Comment (ozandogrultan) [reply frontend/amundsen_application/static/js/components/ResourceListItem/TableListItem/index.tsx]: Thanks for the heads up! I've fixed this.

File: frontend/amundsen_application/static/css/_list-group.scss [hunk 2]
```diff
@@ -16,5 +16,9 @@
       cursor: pointer;
       z-index: 1;
     }
+
+    &.disabled {
```
Comment (Golodhros) [inline frontend/amundsen_application/static/css/_list-group.scss:20]:
```suggestion
    &.is-disabled {
```

File: frontend/amundsen_application/static/js/components/ResourceListItem/TableListItem/index.tsx [hunk 3]
```diff
@@ -34,8 +35,9 @@ const TableListItem: React.FC<TableListItemProps> = ({
   table,
   logging,
   tableHighlights,
+  disabled,
 }) => (
-  <li className="list-group-item clickable">
+  <li className={`list-group-item ${disabled ? 'disabled' : 'clickable'}`}>
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/ResourceListItem/TableListItem/index.tsx:40]:
```suggestion
  <li className={`list-group-item ${disabled ? 'is-disabled' : 'clickable'}`}>
```

**Review by Golodhros — CHANGES_REQUESTED** [review]

File: frontend/amundsen_application/static/js/pages/TableDetailPage/LineageList/index.tsx [hunk 4]
```diff
@@ -16,27 +17,37 @@ export interface LineageListProps {
...
+            logging={logging}
+            key={`lineage-item::${index}`}
+            tableHighlights={getHighlightedTableMetadata(tableResource)}
+            disabled={disabled}
+          />
+        );
+      })}
+    </div>
+  );
+};
```
[the remark this answers is not in the document]
Comment (ozandogrultan) [reply frontend/amundsen_application/static/js/pages/TableDetailPage/LineageList/index.tsx]: I've added some test here. Can you please check again?

File: frontend/amundsen_application/static/js/pages/TableDetailPage/LineageList/index.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          name: 'test_table1',
+          parent: null,
+          schema: 'test_schema',
+          source: 'hive',
+          usage: 0,
+        },
+      ],
+      direction: 'both',
+    };
+    wrapper = shallow<typeof LineageList>(<LineageList {...props} />);
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/pages/TableDetailPage/LineageList/index.spec.tsx:29]: Thanks for the tests!

However, this format of tests with 'wrapper' as a let and using 'shallow' is something legacy we want to move out of.

Check this file https://github.com/amundsen-io/amundsen/blob/7d8088429b224958cc212f9aa4d0fca2f3ead655/frontend/amundsen_application/static/js/components/Alert/index.spec.tsx#L11 for a more up to date pattern for our tests.

Comment (ozandogrultan) [reply frontend/amundsen_application/static/js/pages/TableDetailPage/LineageList/index.spec.tsx]: Thanks for pointing out, I've updated the tests. Please check again.

**Review by Golodhros — APPROVED** [review]
LGTM

**Comment by ozandogrultan** [comment]
@Golodhros @feng-tao Can we merge this if there are no concerns? We can start using the configuration in our app then 🙂

**Comment by Golodhros** [comment]
Waiting for @allisonsuarez to take a look.

**Pull request merged**